src/child.rs

```diff
@@ -74,18 +74,20 @@ macro_rules! delegate {
 /// # }
 /// ```
 #[derive(Debug)]
-pub struct RemoteChild<'s> {
-    session: &'s Session,
+pub struct Child<T> {
+    session: T,
     imp: RemoteChildImp,
 
     stdin: Option<ChildStdin>,
     stdout: Option<ChildStdout>,
     stderr: Option<ChildStderr>,
 }
 
-impl<'s> RemoteChild<'s> {
+pub type RemoteChild<'a> = Child<&'a Session>; 
+
+impl<T> Child<T> {
```

Comment:
```suggestion
impl<S: AsRef<Session> + Clone> Child<S> {
```

Reply:
With the `Clone` bound, you can also implement `fn session() -> S`.

Reply:
Yeah I figured that afterwards but didn't have the time to update the PR.

I don't think the `AsRef` bound is useful since at the end fo the day the `Client` doesn't actually care about the `Session`, beyond the outstanding reference in the borrow case. And the `Clone` case is only useful for `Session` itself.

I guess I could bound the entire thing anyway to ensure `Child` is only used with a session-adjacent object, but it feels unnecessary?

Reply:
That's true, so it's probably ok for `Child` to not constraint `S` since it is not used.